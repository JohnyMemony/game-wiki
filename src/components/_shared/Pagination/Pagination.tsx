import React from 'react';
import { Pagination as EgPagination } from 'evergreen-ui';

interface Props {
  page: number;
  total: number;
  onChange(page: number): void;
}

export const Pagination: React.FC<Props> = (props) => {
  const { page, total, onChange } = props;

  return (
    <EgPagination page={page} totalPages={total} onPageChange={onChange} />
  );
};
