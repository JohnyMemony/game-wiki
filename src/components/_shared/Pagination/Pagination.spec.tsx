import { shallow } from 'enzyme';
import React from 'react';
import { Pagination } from './Pagination';

describe('Pagination component', () => {
  it('should render Pagination component', function () {
    const component = shallow(<Pagination page={1} total={10} onChange={() => {}} />);

    expect(component).toMatchSnapshot();
  });
});