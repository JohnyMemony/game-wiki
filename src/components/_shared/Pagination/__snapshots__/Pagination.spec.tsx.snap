// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component should render Pagination component 1`] = `
<Memo(ForwardRef(Pagination))
  onPageChange={[Function]}
  page={1}
  totalPages={10}
/>
`;
